app/experimenter/experiments/models.py

```diff
@@ -309,6 +312,21 @@ def _get_targeting_max_version(self):
 
         return expressions
 
+    def _get_targeting_pref_conflicts(self):
+        prefs = []
+
+        if self.prevent_pref_conflicts:
+            app_features = {
+                feature.slug: feature
+                for feature in Features.by_application(self.application)
+            }
```

Comment:
Can't you use the `find` method you added below for this?

Reply:
many find() calls vs 1 dict construction